src/service/xhr-impl.js

```diff
@@ -195,9 +176,11 @@ export class Xhr {
    *
    * @param {string} input
    * @param {?FetchInitDef=} opt_init
+   * @param {boolean=} opt_skipProcessing Disables any processing of the
+   *     response, including asserting the response is 2XX.
    * @return {!Promise<!JSONType>}
    */
-  fetchJson(input, opt_init) {
+  fetchJson(input, opt_init, opt_skipProcessing) {
```

Comment:
I think that always returning the response is better than flipping between them. How often do we process errors anyway though? The current signature is basically: "don't care about exact errors - just get me the JSON if you can". How often is this really not the case and shouldn't we just be able to get that data from the `catch` branch?